python/feldera/enums.py

```diff
@@ -1,33 +1,177 @@
 from enum import Enum
```

Comment:
I don't see as much the need of adding extra code specifically to document the enumeration variants. They can just be specified in the docstring of the enumeration as a list like:
```
- SERVER_DEFAULT: ...
- DEV: ...
- UNOPTIMIZED: ...
- OPTIMIZED: ...
```
Which would still be nearby (e.g., when someone adds/edits an enumeration, editing the corresponding documentation is nearby), and the enumerations will likely not see much change over time.